c7n/resources/ec2.py

```diff
@@ -2183,3 +2183,80 @@ class resource_type(query.TypeInfo):
         filter_name = 'ReservedInstancesIds'
         filter_type = 'list'
         arn_type = "reserved-instances"
+
+
+@resources.register('ec2-host')
+class DedicatedHost(query.QueryResourceManager):
+    """Custodian resource for managing EC2 Dedicated Hosts.
+    """
+
+    class resource_type(query.TypeInfo):
+        service = 'ec2'
+        name = id = 'HostId'
+        enum_spec = ('describe_hosts', 'Hosts', None)
+        arn_type = "dedicated-host"
+        filter_name = 'HostIds'
+        filter_type = 'list'
+        date = 'AllocationTime'
+        cfn_type = config_type = 'AWS::EC2::Host'
+        permissions_enum = ('ec2:DescribeHosts',)
+
+
+@DedicatedHost.filter_registry.register('instances-running')
+class InstancesRunning(ValueFilter):
+    """Filter the quantity of instances that are running in the Dedicated Host
+
+    :Example:
+
+    Find hosts that have 0 running instances.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: ec2-host-no-running-instances
+            resource: ec2-host
+            filters:
+              - type: instances-running
+                op: eq
+                value: 0
+    """
+    schema = type_schema('instances-running',
+        required=['op', 'value'],
+        rinherit=ValueFilter.schema,
+        **{
+            'value': {
+                'type': 'integer'
+            }
+        }
+    )
+    schema_alias = False
+    permissions = ('ec2:DescribeHosts',)
+    annotation = 'c7n:InstancesRunning'
+
+    def validate(self):
+        if 'op' not in self.data and 'op' in self.required_keys:
```

Comment:
we can leave off on stuff that's covered by the schema in this case because we're exposing a much more restricting set of matching then the value filter it might be simpler to just express the schema here directly for this filter.

actually looking over the implementation its a bit unclear if we even need this filter, if instances on dedicated hosts is just exposed as an attribute of the host itself.

ie. this one should cover just using the builtin value filter

```
type: value
key: Instances
value_type: value_count
op: equal
value: 0
```

Reply:
Done!

It's good to go, now :)